jlm/llvm/ir/Linkage.hpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2018 Nico Reißmann <nico.reissmann@gmail.com>
+ * See COPYING for terms of redistribution.
+ */
+
+#ifndef JLM_LLVM_IR_LINKAGE_HPP
+#define JLM_LLVM_IR_LINKAGE_HPP
+
+#include <jlm/util/common.hpp>
+
+namespace jlm::llvm
+{
+
+/**
+ * Types of linkage for global variables, constants and functions.
+ * Based on LLVM's "::llvm::GlobalValue::LinkageTypes"
+ */
+enum class Linkage
+{
+  external_linkage,
+  available_externally_linkage,
+  link_once_any_linkage,
+  link_once_odr_linkage,
+  weak_any_linkage,
+  weak_odr_linkage,
+  appending_linkage,
+  internal_linkage,
+  private_linkage,
+  external_weak_linkage,
+  common_linkage
```

Comment:
Since you are touching all the linkage values anyway, you could also just convert them to camelBack. Then we would be done with them as well.

Reply:
done